### PR TITLE
Environment vars / samples config

### PR DESCRIPTION
**Pull request by aarongranick-okta**
Solves many issues caused by config variables stored in multiple places. On a clean install these vars can (are) set by running the tests with environment vars. This allows our CI environment to install and test using only environment variables. However, on the next run the values stored in the config will be used instead of environment variables. This is frustrating and confusing to the local developer.  It is unclear how to recreate the experience of running the samples using environment variables. The problems are multiplied if the developer attempts to run tests.  This PR consolidates these variables to so they are either defined in the shell as env vars or persisted in a single file: `testenv`

- removes `postinstall` step and `default-config.ts` files
- no more editing of `.samples.config.ts` (file is unused now)
- requires ISSUER and CLIENT_ID values to be set as environment variables
 - throws if required environment vars are not set
- uses support for a `testenv` file to persist environment values
- single `testenv` file is shared for all samples in this repo, also used (optionally) for manual test runs
- uses webpack to expose environment vars to Angular

RELATED PR:
https://github.com/okta/samples-nodejs-express-4/pull/69

When the `resource-server` will accept environment variables we will no longer need to run `updateConfig()` in `setup-env.js`

### CODE REVIEW

File: custom-login/extra-webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+const webpack = require('webpack');
+const env = {};
+
+// List of environment variables made available to the app
+[
+  'ISSUER',
+  'CLIENT_ID',
+].forEach(function (key) {
+  if (!process.env[key]) {
+    throw new Error(`Environment variable ${key} must be set. See README.md`);
```
Comment (swiftone) [inline custom-login/extra-webpack.config.js]: Since you've told them to put these in a file, would the mention of environment variables be confusing?

Comment (aarongranick-okta) [reply custom-login/extra-webpack.config.js]: I think I will change the wording in the README so it's more clear they CAN put the environment variables in a file. Some developers would choose to not do this, particularly if they are working with more than one clientId or issuer.

File: scripts/startResourceServer.js [hunk 2]
```diff
@@ -19,9 +19,20 @@ function startResourceServer() {
...
-  exec(command, {maxBuffer: 1024 * 2000}, (err, stdout) => {
+  const command = `node ${dir}/resource-server/server.js`;
+  const options = {
+    env: {
+      ISSUER: process.env.ISSUER,
+      SPA_CLIENT_ID: process.env.CLIENT_ID,
+      PATH: process.env.PATH
+    }
+  };
+  exec(command, options, (err, stdout) => {
```
Comment (swiftone) [inline scripts/startResourceServer.js]: Is the missing maxBuffer significant?  Do we know why that was there?  (Just a Java thing?)

Comment (aarongranick-okta) [reply scripts/startResourceServer.js]: I am assuming it was a Java thing.

**Comment by aarongranick-okta** [comment]
Note that passing environment vars to the resource-server in the `startResourceServer.js` has no effect until related PR is merged, published: https://github.com/okta/samples-nodejs-express-4/pull/69

**Pull request merged**